app/helpers/concerns/foreman_teamdynamix/hosts_helper_extensions.rb

```diff
@@ -1,25 +1,57 @@
 module ForemanTeamdynamix
   module HostsHelperExtensions
     extend ActiveSupport::Concern
-    DEFAULT_TD_PANE_FIELDS = { 'Asset ID': 'ID', 'Owner': 'OwningCustomerName', 'Parent Asset': 'ParentID' }
+    DEFAULT_TD_PANE_FIELDS = { 'Asset ID' => 'ID',
+                               'Owner' => 'OwningCustomerName',
+                               'Parent Asset' => 'ParentID' }.freeze
 
     def teamdynamix_title
       SETTINGS[:teamdynamix][:title] || 'Team Dynamix'
     end
 
-    def teamdynamix_fields(host)
+    def teamdynamix_fields
       td_pane_fields = SETTINGS[:teamdynamix][:fields] || DEFAULT_TD_PANE_FIELDS
-      return [[_('Asset'), 'None Associated']] unless host.teamdynamix_asset_id
-      @asset = host.td_api.get_asset(host.teamdynamix_asset_id)
+      return [[_('Asset'), 'None Associated']] unless @host.teamdynamix_asset_id
+
+      get_teamdynamix_asset(@host.teamdynamix_asset_id)
+
+      # always display a link to the asset
+      fields = [asset_uri]
 
-      fields = []
       td_pane_fields.each do |field_name, asset_attr|
-        asset_attr_val = @asset[asset_attr]
+        asset_attr_val = @asset.key?(asset_attr) ? @asset[asset_attr] : get_nested_attrib_val(asset_attr)
         fields += [[_(field_name.to_s), asset_attr_val]] if asset_attr_val.present?
       end
       fields
     rescue StandardError => e
       [[_('Error'), e.message]]
     end
+
+    private
+
+    def asset_uri
+      api_url = SETTINGS[:teamdynamix][:api][:url]
+      asset_uri = api_url.split('api').first + @asset['Uri']
```

Comment:
I'm not sure its good practice to have a var that is the same name as a method or the method it is actually in. Even if this works it is confusing. Please change the var or the name of the method so they are not the same.